scripts/git/release.sh

```diff
@@ -0,0 +1,136 @@
+
+#!/usr/bin/env bash -e
+VERSION=$1
+ACCESS_TOKEN=$2
+DOCKER_USER=$3
+DOCKER_PWD=$4
+FROM_DOCKERHUB_NAMESPACE=${5:-azureiotpcs}
+TO_DOCKERHUB_NAMESPACE=${6:-azureiotpcs}
+SOURCE_TAG="${7:-testing}"
+DESCRIPTION=$8
+PRE_RELEASE=${9:-false}
+LOCAL=${10}
+APP_HOME="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && cd .. && cd .. && pwd )/"
+
+NC="\033[0m" # no color
+CYAN="\033[1;36m" # light cyan
+YELLOW="\033[1;33m" # yellow
+RED="\033[1;31m" # light red
+
+failed() {
+    SUB_MODULE=$1
+    echo -e "${RED}Cannot find directory $SUB_MODULE${NC}"
+    exit 1
+}
+
+check_input() {
+    if [ ! -n "$VERSION" ]; then
+        echo -e "${RED}Version is required parameter${NC}"
+        exit 1
+    elif [ ! -n "$ACCESS_TOKEN" ]; then
+        echo -e "${RED}Acess_token is required parameter${NC}"
+        exit 1
+    elif [ ! -n "$DOCKER_USER" ]; then
+        echo -e "${RED}Docker username is required parameter${NC}"
+        exit 1
+    elif [ ! -n "$DOCKER_PWD" ]; then
+        echo -e "${RED}Docker password is required parameter${NC}"
+        exit 1
+    fi
+    echo $DOCKER_PWD | docker login -u $DOCKER_USER --password-stdin
+}
+
+tag_build_publish_repo() {
+    SUB_MODULE=$1
+    REPO_NAME=$2
+    DOCKER_CONTAINER_NAME=${3:-$2}
+    DESCRIPTION=$4
+
+    echo
+    echo -e "${CYAN}====================================     Start: Tagging the $REPO_NAME repo     ====================================${NC}"
+    echo
+    echo -e "Current working directory ${CYAN}$APP_HOME$SUB_MODULE${NC}"
+    echo
+    cd $APP_HOME$SUB_MODULE || failed $SUB_MODULE
+
+    if [ -n "$LOCAL" ]; then
+        echo "Cleaning the repo"
+        git reset --hard origin/master
+        git clean -xdf
+    fi
+    git checkout master
+    git pull --all --prune
+    git fetch --tags
+
+    git tag --force $VERSION
+    git push https://$ACCESS_TOKEN@github.com/Azure/$REPO_NAME.git $VERSION
+
+    echo
+    echo -e "${CYAN}====================================     End: Tagging $REPO_NAME repo     ====================================${NC}"
+    echo
+
+    echo
+    echo -e "${CYAN}====================================     Start: Release for $REPO_NAME     ====================================${NC}"
+    echo
+
+    DATA="{
+        \"tag_name\": \"$VERSION\",
+        \"target_commitish\": \"master\",
+        \"name\": \"$VERSION\",
+        \"body\": \"$DESCRIPTION\",
```

Comment:
we might want some validation to ensure $DESCRIPTION doesn't break the json format

Reply:
The description is a string that is in quotes. If somebody passes a json that will be treated as a string. Long term the goal is for this to be a path to RELEASE.md